### ISSUE TITLE
feature request: `res.on('finish', callback)`

### ISSUE THREAD

**Issue by domdomegg** [fastly/expressly#30]
Adding a callback when the request has finished would be useful for tidying up any post-request actions (e.g. sending logs/metrics/traces).

Currently we're working around this using a polling approach on `res.hasEnded`, but this is quite wasteful (all of the polling effort, added complexity and that after the request finishes but before the next poll the app will be sitting doing nothing).

**Comment by rdmulford** [comment]
Agreed this would be useful, looks like @doramatadora is the main contributor here maybe they could help/provide some insight?

**Comment by doramatadora** [comment]
I'm sorry this one took a while–we aim to do better! You can now use [res.on('finish', cb)](https://expressly.edgecompute.app/docs/handling-data/response#resonfinish-callback) in expressly@2.3.0
